BrainPortal/app/models/cluster_task.rb

```diff
@@ -1662,9 +1663,14 @@ def unarchive_work_directory_from_userfile
     end
 
     self.addlog("Attempting to restore TaskWorkdirArchive.")
-
     taskarch_userfile.sync_to_cache
 
+    md5 = self.meta[:workdir_archive_hash]
+    if md5.present? and md5 != taskarch_userfile.cache_compute_md5sum  #  presence check is compatibility layer for old tasks archives
+      self.addlog("MD5 hash does not match stored record: TaskWorkdirArchive #{self.name} is corrupted.")
```

Comment:
In the message, you use `self.name` (the task's name) instead of `taskarch_userfile.name` (the archive's name)